x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -105,13 +104,21 @@ private Item(String id) {
          * Read from a stream.
          */
         public Item(StreamInput in) throws IOException {
-            index = in.readString();
+            if (in.getTransportVersion().onOrAfter(OPTIONAL_INDEX_IN_DOCS_VERSION)) {
+                index = in.readOptionalString();
+            } else {
+                index = in.readString();
+            }
             id = in.readString();
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(index);
+            if (out.getTransportVersion().onOrAfter(OPTIONAL_INDEX_IN_DOCS_VERSION)) {
+                out.writeOptionalString(index);
+            } else {
+                out.writeString(index);
+            }
```

Comment:
Below in the `toXContent` method, we should not write the `INDEX_FIELD` if `this.index == null`.

Reply:
Done, thanks!